app/service/planning_svc.py

```diff
@@ -58,11 +65,17 @@ def __init__(self):
         :return: None
         """
         for member in operation['host_group']:
-            op = await self.get_service('data_svc').explode_operation(dict(id=operation['id']))
+            if (not member['trusted']) and (not operation['allow_untrusted']):
+                continue
+            op = await self.get_service('data_svc').explode_operation(criteria=dict(id=operation['id']))
             while next((True for lnk in op[0]['chain'] if lnk['paw'] == member['paw'] and not lnk['finish']),
                        False):
                 await asyncio.sleep(3)
-                op = await self.get_service('data_svc').explode_operation(dict(id=operation['id']))
+                if not operation['allow_untrusted']:
```

Comment:
let's break this logic into a private function. so in this function, it would look like:
```
                if await self._trust_issues(operation, member['paw']):
                    break
```
then in a private function, you would have:
```
    async def _trust_issues(self, operation, paw):
        if not operation['allow_untrusted']:
            agent = await self.get_service('data_svc').explode_agents(criteria=dict(paw=paw))
            return not agent[0]['trusted']
        return False
```